packages/react-app/src/views/BuildsListView.jsx

```diff
@@ -1,12 +1,42 @@
 import React, { useEffect, useState } from "react";
-import { Container, SimpleGrid, Heading, Text } from "@chakra-ui/react";
+import { useUserAddress } from "eth-hooks";
+import {
+  useToast,
+  useColorModeValue,
+  Container,
+  Button,
+  Input,
+  Textarea,
+  FormControl,
+  FormLabel,
+  FormErrorMessage,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalCloseButton,
+  ModalHeader,
+  ModalOverlay,
+  SimpleGrid,
+  Heading,
+  Text,
+} from "@chakra-ui/react";
 import BuildCard from "../components/BuildCard";
-import { getAllBuilds } from "../data/api";
+import { getAllBuilds, getBuildSubmitSignMessage, postBuildSubmit } from "../data/api";
 import useCustomColorModes from "../hooks/useCustomColorModes";
 
-export default function BuildsListView() {
+export default function BuildsListView({ userProvider }) {
   const [builds, setBuilds] = useState([]);
+  const [isSubmitModalOpen, setIsSubmitModalOpen] = useState(false);
```

Comment:
Just FYI, Chakra has a fancy useDisclosure hook to handle open / close / toggle scenarios.

https://chakra-ui.com/docs/hooks/use-disclosure

Reply:
Awesome. I just found two usages in our app, so I'll update the code for consistency.